### PR TITLE
feat and tests : transferTokensERC20

### PR DESCRIPTION
**Pull request by 10priyanshudhiman**
ERC20 Transfer Tokens
1. Added transferTokens where escrow can perform transfer
2. Added updateTotalBalance which is used to change the state varible totalBalance which is escrow platfrom balance
3. Added test for transferTokens where escrow does not need to approve to player for transfer as he can directly send them.
4. Added test for if the platform balance is zero or that cantractBalance whose tokens are is zero.

### CODE REVIEW

**Review by rahulghangas — DISMISSED** [review]

File: src/EscrowERC20.sol [hunk 1]
```diff
@@ -28,6 +29,7 @@ contract EscrowERC20 {
     address public escrAddress;
     address public playerAddress;
     address public contractAddress;
+    address public adminAddress = 0x4cd4df5E4485ffd09345bB5dAC0fcE06Dd00ef07;
```
Comment (shryasss) [inline src/EscrowERC20.sol:32]: This should be set in constructor.

File: src/EscrowERC20.sol [hunk 2]
```diff
@@ -53,6 +55,18 @@ contract EscrowERC20 {
         playerToContractToAmount[playerAddress][contractAddress] = _Amount;
     }
 
+    function transferTokens(address _PlayerAddress, address _ContractAddress, uint256 _Amount)
+        public
+        inEscrAvailable(EscrERC20Available.YES)
```
Comment (shryasss) [inline src/EscrowERC20.sol:60]: remove this

File: src/EscrowERC20.sol [hunk 3]
```diff
@@ -53,6 +55,18 @@ contract EscrowERC20 {
         playerToContractToAmount[playerAddress][contractAddress] = _Amount;
     }
 
+    function transferTokens(address _PlayerAddress, address _ContractAddress, uint256 _Amount)
+        public
```
Comment (shryasss) [inline src/EscrowERC20.sol:59]: This should be external.

File: src/EscrowERC20.sol [hunk 4]
```diff
@@ -53,6 +55,18 @@ contract EscrowERC20 {
...
+        inEscrAvailable(EscrERC20Available.YES)
+        onlyAdmin
+    {
+        playerAddress = _PlayerAddress;
```
Comment (shryasss) [inline src/EscrowERC20.sol:63]: Use the variables directly instead of assigning to state variables.

File: src/EscrowERC20.sol [hunk 5]
```diff
@@ -96,4 +117,8 @@ contract EscrowERC20 {
         }
         return playerToContractToAmount[_PlayerAddress][_ContractAddress];
     }
+
+    function updateTotalBalance(uint256 newBalance) public {
```
Comment (shryasss) [inline src/EscrowERC20.sol:121]: Write comments for functions written for test purposes.

**Review by shryasss — CHANGES_REQUESTED** [review]
Please do the suggested changes.

File: src/EscrowERC20.sol [hunk 6]
```diff
@@ -3,88 +3,101 @@
...
 
-    EscrERC20Available public escrAvailable;
+    bool public escrAvailable;
     mapping(address => mapping(address => uint256)) playerToContractToAmount;
     mapping(address => address[]) playerToContracts;
 
     constructor() {
-        escrAddress = address(this);
-        escrAvailable = EscrERC20Available.YES;
+        adminAddress = 0x4cd4df5E4485ffd09345bB5dAC0fcE06Dd00ef07;
```
Comment (shryasss) [inline src/EscrowERC20.sol:30]: Do not hardcode admin address, instead send it as a argument to the constructor.

**Review by shryasss — CHANGES_REQUESTED** [review]
Requested minor changes

**Review by shryasss — APPROVED** [review]
LGTM

**Review by rahulghangas — APPROVED** [review]

**Pull request merged**